rfc-182-flexible-pages.md

```diff
@@ -0,0 +1,176 @@
+
+The plan is to create a minimal viable product (MVP) of flexible pages that will allow us to use them to recreate an existing page type. Further work will then be done to expand and improve flexible pages beyond MVP.
+
+## Proposal
+
+One of the key considerations when deciding upon an approach for flexible pages is the amount of flexibility that is given to the user (content designers working in GDS and government departments). This can be thought of as a scale of flexibility, with two extreme ends:
+
+- Full control: the user has tools available to them that will allow full customisation of layout and page content with almost no restrictions.
+- Limited control: the user would choose from different preset page layouts, but only enter content in very restricted ways.
+
+Neither option is desirable. Limited control would defeat the point of a ‘flexible page’. Full control could lead to a loss of consistency across GOV.UK content, potentially create accessibility issues, and create a system that would be extremely difficult to edit and manage.
+
+Instead, we propose a middle ground, where users cannot control page layout but can construct pages from a series of pre-built flexible sections, into which specific content can be added. This will help to preserve and protect the GOV.UK brand, the consistency of GOV.UK pages and their accessibility.
+
+### Flexible sections
+
+Flexible sections occupy the full width of the page and would never occur visibly side by side. They contain a layout based on existing GOV.UK layouts (for example a two thirds/one third layout) but allow no control over that layout.
+
+A flexible section should be designed for a particular use and only allow content within its layout based on that use. Users should be able to choose a flexible section and add content into it, but have little to no control over how that content is positioned. Users should be able to change the order of flexible sections within a page.
+
+Flexible sections should be specific enough to provide an understandable ‘building block’ for making a page, without being generic enough to allow users free reign - they should have a specific purpose. For example, a flexible section that only defines layout would allow too much freedom to put content anywhere within it.
+
+Examples of flexible sections could be:
+
+- a page title flexible section, limited to a two thirds column, that contains the page H1 and a lead paragraph
+- a govspeak flexible section, with a one third column containing a contents list and a two thirds column containing govspeak and images
+
+Initially we propose only building the flexible sections needed for MVP. Further flexible sections or variations of flexible sections would be constructed as the work on flexible pages continues. For example, an additional page title flexible section could be built to provide the page title but in a GOV.UK blue box, or this could be provided as an option on the existing page title flexible section.
```

Comment:
All of the above looks great. And from a Publishing perspective, iterating on existing 'flexible sections' - rather than adding new 'flexible sections' when the tiniest deviation is needed - is also a good way forward. We have experience of [taking a pre-agreed schema and then dynamically rendering a form for said schema](https://github.com/alphagov/specialist-publisher/pull/2929), and we have experience of [iterating the schema and form to allow for new features](https://github.com/alphagov/specialist-publisher/pull/3049).